### ISSUE TITLE
IdentityServer with openid with Ajax call

### ISSUE THREAD

**Issue by tlp0308** [IdentityServer/IdentityServer3#3376]
Hi all,
I'm developing microservices system, and I'm using identityserver3. My system is designing like this
. MVC Application. 
. WebAPI 

The user needs to log in from MVC application, from there MVC controller will make a call to web API to operate the business.
MVC application just renders the view, in the view, I'm using ajax to call **MVC controller and MVC controller will call API.** I will get the access token in MVC controller when calling Web API.
Recently, I encounter the problem that after a period or API throws an error. I can't get the access token anymore. I need to re-log in again.
After researching, I put the option UseTokenLifetime= false and its working but not perfectly. Sometimes, I encounter the unauthorized error.
This is my configuration for Web MVC.

` app.UseOpenIdConnectAuthentication(new OpenIdConnectAuthenticationOptions'
            {
                Authority = "https://xxxx/core",
                ClientId = "etonops",
                Scope = "openid profile ",
                ResponseType = "id_token token",
                RedirectUri = "xxx",
                SignInAsAuthenticationType = "Cookies",
                UseTokenLifetime = false,
`
This is the ajax code from mvc views that is using to call mvc controller.
`$.ajax({
            url: "/Aisle/GetDataBeforeCreate",
            data: {},
            type: 'POST',
            success: function (data) {
                if (data !== null) {`
and Web api configuration.

`app.UseIdentityServerBearerTokenAuthentication(new IdentityServerBearerTokenAuthenticationOptions
            {
                Authority = "https://xxxx/core",
                RequiredScopes = new[] { "etonopsservice" },

            });`

My questions here:
- If I call the MVC controller from ajax. Do I need to put the access token? Currently, I can call the MVC application controller (with authorized attribute) without Ajax call header. It's weird
- Am I missing something about configuration?
I'm really appreciate your help.
Thanks,
CK

**Comment by brockallen** [comment]
This seems to be a general question about IdentityServer - not a bug report or an issue.

Please use StackOverflow for that. This has the advantage that questions and answers can be easily found by search engines, and that there are more people answering questions than just us.

For IdentityServer3
https://stackoverflow.com/questions/tagged/?tagnames=identityserver3&sort=newest

For IdentityServer4
https://stackoverflow.com/questions/tagged/?tagnames=identityserver4&sort=newest

For commercial support
https://identityserver.io/
